sway/input/seat.c

```diff
@@ -686,41 +703,20 @@ void seat_set_focus_warp(struct sway_seat *seat,
 
 	// If we've focused a floating container, bring it to the front.
 	// We do this by putting it at the end of the floating list.
-	if (container && container_is_floating(container)) {
-		list_move_to_end(
-				container->parent->sway_workspace->floating, container);
-	}
-
-	// clean up unfocused empty workspace on new output
-	if (new_output_last_ws) {
-		workspace_consider_destroy(new_output_last_ws);
```

Comment:
I'm pretty sure this is needed. To test, have two outputs. On the left output have workspace 1, and on the right output have workspace 2 and 3. Make workspace 2 visible and empty. Focus workspace 1 on the left output. Then focus workspace 3. This code should destroy workspace 2.